vault/resource_consul_secret_backend_role.go

```diff
@@ -257,17 +271,21 @@ func consulSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) error
 		"node_identities":    "node_identities",
 	}
 
-	_, hasLegacyPolicies := data["policies"]
-	if hasLegacyPolicies {
-		params["policies"] = "policies"
-		if _, ok := d.GetOk("consul_policies"); ok {
-			params["policies"] = "consul_policies"
-		}
+	// Check whether Vault will return consul_policies or policies based on its version.
+	hasNewPolicies, _, err := semver.GreaterThanOrEqual(ctx, client, consts.VaultVersion11)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	returnedPoliciesVal := "consul_policies"
```

Comment:
This is the target field to set in the in the state, rather than a value.

Reply:
+1, would suggest renaming to `stateKey` or `field` as done in other resources for explicitness

Reply:
👍